doc/release-notes-33629.md

```diff
@@ -0,0 +1,41 @@
+Mempool
+=======
+
+The mempool has been reimplemented with a new design ("cluster mempool"), to
+facilitate better performance when constructing block templates, evicting
+transactions, relaying transactions, and validating replacement transactions
+(RBF). Most changes should be transparent to users, but some behavior changes
+are noted:
+
+- The mempool no longer enforces ancestor or descendant size/count limits.
+  Instead, two new limits are introduced: `-limitclustercount` (default 64) and
+  `-limitclustersize` (default 101kvB). These limits govern the size of
+  connected components, or clusters, in the mempool.  Transactions are
+  considered to be in the same cluster if they are connected to each other via
+  parent/child relationships in the mempool.
+
+- Within the mempool, transactions are ordered based on the fee rate at which
+  they are expected to be mined, which takes into account the full set, or
+  "chunk", of transactions that would be included together (eg a parent and its
```

Comment:
In commit "doc: add release notes snippet for cluster mempool"

Writing nit: `eg` -> `e.g.,`.

Reply:
Fixed.